### ISSUE TITLE
Update minimum requirement of torch to 1.8

### ISSUE THREAD

**Issue by SkafteNicki** [Lightning-AI/metrics#1262]
## 🚀 Feature

Based on this comment thread https://github.com/Lightning-AI/metrics/pull/1259#discussion_r992212074 we have agreed on updating the minimum requirement of pytorch to be 1.8 which would still mean that `torchmetrics` would be compatible with 5 major versions of pytorch.

### Motivation

We are running into problems with core pytorch functions such as `torch.where` not supporting specific dtypes because the operations were simply not fully implemented in earlier versions. Until now we have taken two approaches to this:
1. Either write wrappers that take care of the problems
    ```python
    def some_function(*args):
        if _PYTORCH_LOWER_THAN_1_8:
            return custom_fn(*args)
        else:
            torch.some_function(*args)
    ```
2. Raise an error and ask users to update if they want to use the specific metric/feature

### Additional context

In addition to the requirement file, we should easily be able to find what needs to changes by looking for these imports:
https://github.com/Lightning-AI/metrics/blob/d3c4c8230c53d037aa05be0094e6206af0ee6d7a/src/torchmetrics/utilities/imports.py#L98-L103
CI workflow files should also be updated.
